pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -366,3 +367,43 @@ func (c *Cluster) deleteAllStatusConfigMaps() {
 		}
 	}
 }
+
+// delete the prepare jobs, which are completed and have duplicated placements
+func (c *Cluster) deletePrepareJobs() {
+	// list all the prepare jobs
+	jobName := "rook-ceph-osd-prepare"
+	listOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", k8sutil.AppAttr, jobName),
+	}
+
+	jobs, err := c.context.Clientset.BatchV1().Jobs(c.clusterInfo.Namespace).List(c.clusterInfo.Context, listOpts)
+	if err != nil {
+		logger.Warningf("failed to list OSD prepare jobs. %v", err)
+		return
+	}
+
+	// filter out the completed jobs
+	for _, job := range jobs.Items {
+		if job.Status.Succeeded == 1 {
+			c.deleteDuplicatePlacementJobs(job)
+		}
+	}
+}
+
+func (c *Cluster) deleteDuplicatePlacementJobs(job batchv1.Job) {
+	// filter the jobs that has duplicated tolearations
+	tolerations := job.Spec.Template.Spec.Tolerations
+
+	for _, tole := range tolerations {
+		for _, tole2 := range tolerations {
```

Comment:
won't it compare the same map to itself every time?

Reply:
Right will update it